src/main/java/com/google/cloud/teleport/templates/PubsubToText.java

```diff
@@ -140,37 +223,104 @@ public static PipelineResult run(Options options) {
     // Create the pipeline
     Pipeline pipeline = Pipeline.create(options);
 
-    /*
-     * Steps:
-     *   1) Read string messages from PubSub
-     *   2) Window the messages into minute intervals specified by the executor.
-     *   3) Output the windowed files to GCS
-     */
-    pipeline
-        .apply("Read PubSub Events", PubsubIO.readStrings().fromTopic(options.getInputTopic()))
-        .apply(
-            options.getWindowDuration() + " Window",
-            Window.into(FixedWindows.of(DurationUtils.parseDuration(options.getWindowDuration()))))
-
-        // Apply windowed file writes. Use a NestedValueProvider because the filename
-        // policy requires a resourceId generated from the input value at runtime.
-        .apply(
-            "Write File(s)",
-            TextIO.write()
-                .withWindowedWrites()
-                .withNumShards(options.getNumShards())
-                .to(
-                    new WindowedFilenamePolicy(
+    String format = options.getFormat();
+
+    switch (format) {
```

Comment:
Note that the value of the format field is determined at graph creation time. i.e. when this template is compiled and stored on GCS. The "runtime" value of this field is ignored and so this template would essentially either follow the "string" path or the "json" path depending on the value passed to --format at compile time. This would mean that this pipeline would work as a normal dataflow job but not as a template (or you need two different templates compiled with different values for --format)

In Dataflow templates, the job graph does not change after it has been compiled. Take a look at the [PubsubToBigQuery template](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/master/src/main/java/com/google/cloud/teleport/templates/PubSubToBigQuery.java) as an example. It uses a boolean flag called --useSubscription. This is used to create two different pipelines in the Dataflow UI ("Cloud Pub/Sub subscription to BigQuery" and "Cloud Pub/Sub topic to BigQuery")

Reply:
I'm second to @sabhyankar. Because of the limitation of the templates feature, this has to be a separate template in our dropdown list. We're working on extending the feature but it may need some time. It'll fit into "several weeks" time frame.

Unfortunately before that you'll have to generate variants of templates on your side.